### PR TITLE
Refactor(Scripts/Naxxramas): Spell Scripts use registry macros

### PR DESCRIPTION
**Pull request by sogladev**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

only refactor

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

compiled, ran worldserver

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Kitzunu — COMMENTED** [review]
All spells used within a spellscript should be validated at the top of the script see https://www.azerothcore.org/wiki/core-scripts#validation-of-spells-used-in-a-script

```cpp
    bool Validate(SpellInfo const* /*spellInfo*/) override
    {
        return ValidateSpellInfo({ SPELL_1, SPELL_2 });
    }
```

File: src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp [hunk 1]
```diff
@@ -606,121 +606,99 @@ class boss_thaddius_summon : public CreatureScript
...
-                GetCaster()->SetAuraStack(spellId, GetCaster(), count);
-            }
         }
 
-        void HandleDamage(SpellEffIndex /*effIndex*/)
+        if (count)
         {
-            if (!GetTriggeringSpell())
-                return;
+            uint32 spellId = GetSpellInfo()->Id == SPELL_POSITIVE_CHARGE ? SPELL_POSITIVE_CHARGE_STACK : SPELL_NEGATIVE_CHARGE_STACK;
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp:632]: need to validate :)

**Comment by sogladev** [comment]
ci does not trigger when marking draft->review, perhaps it's worth adding `ready_for_review` to https://github.com/azerothcore/azerothcore-wotlk/blob/d77b3ac110d03c57459affc5ea81aa7fad1fab59/.github/workflows/core-build-nopch.yml#L10  

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

File: src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp (hunk 1 above)
Comment (sogladev) [reply src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp:632]: Even if the spell is not used/casted? It's just an enum

Comment (Kitzunu) [reply src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp:637]: it is used
```cpp
GetCaster()->SetAuraStack(spellId, GetCaster(), count);
```

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**